src/host/os_getnumcpus.c

```diff
@@ -0,0 +1,88 @@
+/**
+ * \file   os_getnumcpus.c
+ * \brief  Retrieve the logical number of CPUs of the host system.
+ * \author Copyright (c) 2002-2024 Jason Perkins and the Premake project
+ */
+
+#if __linux__
+#define _GNU_SOURCE
+#endif
+
+#include "premake.h"
+
+#if PLATFORM_LINUX
+#include <sched.h>
+#elif PLATFORM_SOLARIS | PLATFORM_AIX | PLATFORM_MACOSX | PLATFORM_BSD
+#include <sys/sysctl.h>
+#endif
+
+int do_getnumcpus()
+{
+#if PLATFORM_WINDOWS
```

Comment:
We should explicitly include the windows header. Don't rely on transient includes.